keygen/src/keygen.rs

```diff
@@ -448,6 +460,15 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                         .long("use-mnemonic")
                         .help("Generate using a mnemonic key phrase.  Expect a significant slowdown in this mode"),
                 )
+                .arg(
+                    Arg::with_name("use_derivation_path")
+                        .long("use-derivation-path")
```

Comment:
Dropping the "use" from the argument name seems fine, so it's just `--derivation-path`
```suggestion
                    Arg::with_name("derivation_path")
                        .long("derivation-path")
```

Reply:
I tried this option, but chose with `use`.
With value it looks good, but without value (when default) it looks unclear.

Compare:
```
solana-keygen new --derivation-path -o keypair.json
solana-keygen new --use-derivation-path -o keypair.json

solana-keygen new --derivation-path m/44/2017/0/1 -o keypair.json
solana-keygen new --use-derivation-path m/44/2017/0/1 -o keypair.json

solana-keygen grind --use-mnemonic --derivation-path --starts-with a:1
solana-keygen grind --use-mnemonic --use-derivation-path --starts-with a:1

solana-keygen grind --use-mnemonic --derivation-path m/44/2017/0/1 --starts-with a:1
solana-keygen grind --use-mnemonic --use-derivation-path m/44/2017/0/1 --starts-with a:1
```

What will be your decision?

Reply:
But maybe no decision is needed. See what Nelson has to offer https://github.com/solana-labs/solana/pull/21614#discussion_r782606765